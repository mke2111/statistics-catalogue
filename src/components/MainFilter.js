import React from 'react';
import { PropTypes } from 'prop-types';
import mainFilterCat from '../config/appConfig';

const MainFilter = ({
  changeMainFilter,
}) => {
  const handleChange = (e) => {
    changeMainFilter(e.target.value);
  };

  return (

    <div className="s">
      <select className="categories" onChange={handleChange}>
        {
          mainFilterCat.map((cat, id) => (
            <option key={`opt_${id * 2}`} value={cat.api}>{cat.key}</option>
          ))
        }
      </select>
    </div>
  );
};

MainFilter.propTypes = {
  changeMainFilter: PropTypes.func.isRequired,
};

export default MainFilter;
