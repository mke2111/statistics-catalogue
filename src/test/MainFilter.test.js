import React from 'react';
import renderer from 'react-test-renderer';
import MainFilter from '../components/MainFilter';

describe('Test Display component with snapshot', () => {
  it('Renders Display component correctly', () => {
    const display = renderer.create(<MainFilter />).toJSON();

    expect(display).toMatchSnapshot();
  });
});
