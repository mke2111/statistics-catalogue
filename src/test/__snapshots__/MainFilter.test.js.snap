// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Display component with snapshot Renders Display component correctly 1`] = `
<div>
  <select
    onChange={[Function]}
  >
    <option
      value="market_cap"
    >
      Market Cap
    </option>
    <option
      value="circulating_supply"
    >
      Circulating Supply
    </option>
    <option
      value="ath_change_percentage"
    >
      24h change vs ATH%
    </option>
  </select>
</div>
`;
